### PR TITLE
Added a framework for accurate and efficient simulation of SiDB logic

### PR DESCRIPTION
**Pull request by Drewniok**
✨ Added a framework for accurate and efficient simulation of SiDB logic.

### CODE REVIEW

File: include/fiction/technology/charge_distribution_surface.hpp [hunk 1]
```diff
@@ -0,0 +1,828 @@
...
+
+template <typename Lyt>
+class charge_distribution_surface<Lyt, true> : public Lyt
+{
+  public:
+    explicit charge_distribution_surface(const Lyt& lyt) : Lyt(lyt) {}
+};
+
+template <typename Lyt>
+class charge_distribution_surface<Lyt, false> : public Lyt
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/technology/charge_distribution_surface.hpp:51]: we added the copy assignment constructor, but rule of three?

File: include/fiction/technology/sidb_charge_state.hpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }
+}
+
+/**
+ * Converts the vector of charges (enums) to a string (-101...).
+ *
+ * @param charge_distribution vector of the SiDBs' charge states.
+ * @return string of the charge states.
+ */
+[[nodiscard]] std::string charge_configuration_to_string(const std::vector<sidb_charge_state> &charge_distribution) noexcept
```
[the remark this answers is not in the document]
Comment (Drewniok) [reply include/fiction/technology/sidb_charge_state.hpp:67]: sorry, I forgot to include ``vector``. My fault!

**Review by marcelwa — CHANGES_REQUESTED** [review]

File: include/fiction/algorithms/simulation_sidb/time_to_solution.hpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ * @param charge_distribution_surface<Lyt> charge layout that is used for the simulation.
+ * @param tts_stats struct where the results (time_to_solution, acc, single runtime) are stored.
+ * @param repetitions number of repetitions to determine the simulation accuracy (repetitions = 100 ==> accuracy is
+ * precise to 1 %).
+ * @param iteration_steps simulation parameter (see quicksim.hpp).
+ * @param alpha simulation parameter (see quicksim.hpp).
+ * @param convlevel the time-to-solution also depends one the given confidence level which can be set here.
+ */
+template <typename Lyt>
+void sim_acc_tts(charge_distribution_surface<Lyt>& lyt, exgs_stats<Lyt>& result_exact,
```
Comment (marcelwa) [inline include/fiction/algorithms/simulation_sidb/time_to_solution.hpp:56]: Why is this function calling `quicksim` itself but requires the user to pass `exgs_stats` from a previous run? Can it not call that itself too?

File: include/fiction/algorithms/simulation_sidb/check_groundstate.hpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    if (exhaustive_results.valid_lyts.empty())
+    {
+        return false;
+    }
+
+    const auto min_energy_exact  = minimum_energy(exhaustive_results.valid_lyts);
+    const auto min_energy_new_ap = minimum_energy(quicksim_results.valid_lyts);
+
+    return std::abs(min_energy_exact - min_energy_new_ap) / min_energy_exact < 0.00001;
```
Comment (marcelwa) [inline include/fiction/algorithms/simulation_sidb/check_groundstate.hpp:44]: Didn't you define the `POP_STABILITY_ERR` constant for this?

**Review by marcelwa — APPROVED** [review]

**Pull request merged**